setuptools/build_meta.py

```diff
@@ -97,16 +97,16 @@ def _get_immediate_subdirectories(a_dir):
 
 
 def _file_with_extension(directory, extension):
-    matching = (
+    matching = [
```

Comment:
I saw the conversation that inspired this change, but if I hadn't, I wouldn't understand why this was here and I'd be reluctant to accept the change. Moreover, without that context or seeing this diff, I'd be inclined to change this back to a generator expression to build on lazy evaluation.

I'm slightly inclined to think this code is becoming overly-protective and it may not be worth the effort to try to protect against unexpected value errors or to hide them from the output.